### PR TITLE
adding retail vertex-ai demo to labs repo

### PR DESCRIPTION
**Pull request by valerioarvizzigno**
Added the "Homecraft" retail vertexAI-based genAI demo we will show at Google Next 23. The demo shows how to leverage Google's Palm2 calls to build an intelligent e-commerce search bar. The demo integrates with data sources ingested via webcrawler, raw-file datasets and BigQuery public datasets. Also a fine-tuned version of Palm2 is used for specific task handling.

### CODE REVIEW

**Comment by valerioarvizzigno** [comment]
@joemcelroy dunno if you are the right person to tag here, but saw you approved previous PRs. Is anything missing for this PR to be accepted (I've already signed the contributor agreement)? Please let me know, as requested by Serena Chou I'm requesting this so to have the Retail vertexAI-based RAG Demo officially merged in this repo to be publicly shared at Google Next. thanks

File: .DS_Store
Comment (joemcelroy) [inline .DS_Store:1]: remove this file please!

File: gai-examples/.DS_Store
Comment (joemcelroy) [inline gai-examples/.DS_Store:1]: same here

**Comment by joemcelroy** [comment]
could you please 
1. move this into the example-apps folder
2. add to the readme in example-apps folder talking more about the app

Thanks!

**Comment by valerioarvizzigno** [comment]
@joemcelroy thanks for your review!

I fixed as per your request, I hope I haven't messed up anything on my fork, had to refetch since whole structure of this repo changed. Let me know!

Thanks a lot

**Review by joemcelroy — APPROVED** [review]

**Pull request merged**